libraries/chain/database_worker_objects.cpp

```diff
@@ -155,35 +155,79 @@ namespace golos { namespace chain {
 
         const auto& wto_idx = get_index<worker_techspec_index, by_next_cashout_time>();
 
-        for (auto wto_itr = wto_idx.begin(); wto_itr != wto_idx.end() && wto_itr->next_cashout_time <= now; ++wto_itr) {
-            const auto& wpo = get_worker_proposal(wto_itr->worker_proposal_author, wto_itr->worker_proposal_permlink);
+        // Updating consumption
+
+        asset consumption;
+
+        auto month_sec = fc::days(30).to_seconds();
+        for (auto wto_itr = wto_idx.begin(); wto_itr != wto_idx.end() && wto_itr->next_cashout_time < time_point_sec::maximum(); ++wto_itr) {
```

Comment:
This loop is excess.
You can increase consumption on approving.
And decrease it on disapproving and complete of payments.

Reply:
@afalaleev 

1. Increase consumption on final approving.
And decrease it on complete of payments. No?
2. Is it useful to store consumption of techspec in object, to calculate it only in one place of code?